apps/dashboard/lib/trpc/routers/authorization/permissions/upsert.ts

```diff
@@ -0,0 +1,201 @@
+import { permissionSchema } from "@/app/(app)/authorization/permissions/components/upsert-permission/upsert-permission.schema";
+import { insertAuditLogs } from "@/lib/audit";
+import { and, db, eq, schema } from "@/lib/db";
+import { requireUser, requireWorkspace, t } from "@/lib/trpc/trpc";
+import { TRPCError } from "@trpc/server";
+import { newId } from "@unkey/id";
+
+export const upsertPermission = t.procedure
+  .use(requireUser)
+  .use(requireWorkspace)
+  .input(permissionSchema)
+  .mutation(async ({ input, ctx }) => {
+    const isUpdate = Boolean(input.permissionId);
+    let permissionId = input.permissionId;
+
+    if (!isUpdate) {
+      permissionId = newId("permission");
+      if (!permissionId) {
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "Failed to generate permission ID",
+        });
+      }
+    }
+
+    if (!permissionId) {
+      throw new TRPCError({
+        code: "INTERNAL_SERVER_ERROR",
+        message: "Invalid permission ID",
+      });
+    }
```

Comment:
this seems redundant. The boolean conversion will mark it as a `isUpdating = false` if its falsy, and then `newId` will take care of it. I don't see this check ever triggering